packages/teams-js/src/internal/communication.ts

```diff
@@ -257,6 +264,90 @@ export function sendMessageToParentAsync<T>(
   });
 }
 
+type SimpleType = string | number | boolean | null | undefined | SimpleType[];
+
+function isSerializable(arg: unknown): arg is ISerializable {
+  return (
+    arg !== undefined &&
+    arg !== null &&
+    (arg as ISerializable).serialize !== undefined &&
+    typeof (arg as ISerializable).serialize === 'function'
+  );
+}
+
+function serializeItemArray(items: (SimpleType | ISerializable)[]): unknown[] {
+  return items.map((item) => {
+    if (isSerializable(item)) {
+      return item.serialize();
+    } else {
+      return item;
+    }
+  });
+}
+
+/**
+ * Call a function in the host and receive a response. If the host returns an {@link SdkError} instead of a normal response, this function will throw a new Error containing the SdkError's information
+ *
+ * @param apiVersionTag A unique tag used to identify the API version for telemetry purposes.
+ * @param functionName The function name to call in the host.
+ * @param responseHandler When the host responds, this handler will validate and deserialize the response.
+ * @param args A collection of data to pass to the host. This data must be an array of either simple types or objects that implement {@link ISerializable}.
+ * @param errorChecker This optional property can be used to override the default ErrorChecking this function uses to decide whether to throw the host response as a new Error. Specify this is your function needs to do any logic verifying that the object received is an error that goes beyond the logic found in {@link isSdkError}.
+ *
+ * @returns The response received from the host after deserialization.
+ *
+ * @throws An Error containing the SdkError information ({@link SdkError.errorCode} and {@link SdkError.message}) if the host returns an SdkError, or an Error if the response from the host is an unexpected format.
+ */
+export async function callFunctionInHostAndHandleResponse<ReceivedFromHost, DeserializedFromHost>(
+  apiVersionTag: string,
+  functionName: string,
+  responseHandler: ResponseHandler<ReceivedFromHost, DeserializedFromHost>,
```

Comment:
>ReceivedFromHost [](http://example.com/codeflow?start=35&length=16)

Consider `SerializedReturnValueFromHostFunction` -- I know, it's a bit long, but it's also very clear

Reply:
How about `SerializedReturnValueFromHost`, it removes the `Function` part like jay said and is more clear that what I had originally. I don't have a problem with long type names because autocomplete exists, but the `Function` suffix seems unnecessary to me given the context this is used in